api/envoy/config/trace/v3/skywalking.proto

```diff
@@ -0,0 +1,59 @@
+syntax = "proto3";
+
+package envoy.config.trace.v3;
+
+import "envoy/config/core/v3/grpc_service.proto";
+
+import "google/protobuf/wrappers.proto";
+
+import "udpa/annotations/migrate.proto";
+import "udpa/annotations/sensitive.proto";
+import "udpa/annotations/status.proto";
+import "udpa/annotations/versioning.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.config.trace.v3";
+option java_outer_classname = "SkywalkingProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_migrate).move_to_package =
+    "envoy.extensions.tracers.skywalking.v4alpha";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: SkyWalking tracer]
+
+// Configuration for the SkyWalking tracer. Please note that if SkyWalking tracer is used as the
+// provider of http tracer, then
+// :ref:`start_child_span <envoy_v3_api_field_extensions.filters.http.router.v3.Router.start_child_span>`
+// in the router must be set to true to get the correct topology and tracing data.
+// [#extension: envoy.tracers.skywalking]
+message SkyWalkingConfig {
+  // SkyWalking collector service.
+  core.v3.GrpcService grpc_service = 1 [(validate.rules).message = {required: true}];
+
+  ClientConfig client_config = 2;
+}
+
+// Client config for SkyWalking tracer.
+message ClientConfig {
+  // Service name for SkyWalking tracer. If service_name is empty, then cluster name of Envoy will
```

Comment:
What if it's a Google gRPC service without cluster name?

Reply:
The cluster name here refers to the cluster name in Envoy localinfo (configured by node.cluster in bootstrap or --service-cluster). It is used to indicate the identity of Envoy when reporting data to the skywalking server. I will update this comment to explain it more clearly.

Reply:
@htuch  But you reminded me that the cluster name in localinfo may also be empty. Here I will make a modification. If the cluster name is also empty (which will not happen in most cases), use `EnvoyProxy` as the service name.

Reply:
Yeah, please clarify that you mean "local cluster name". Is `EnvoyProxy` some standard? Is there a reason empty string won't work?

Reply:
@htuch `EnvoyProxy` is not standard but a choice. It is component name of Envoy in the Skywalking (https://github.com/wbpcode/skywalking/blob/129effe4e12211fcccdc40249db896728457dc21/oap-server/server-bootstrap/src/main/resources/component-libraries.yml) So I chose this name as the default name when no any configuration is available. In practice, I have almost never encountered a situation where the local cluster name and local node name are empty, and the control plane always generates these two configurations for Envoy. So I think this default name only exists as an insurance and will not have any chance of being actually used.

Service name and instance name are indispensable when creating the SkyWalking propagation header.

Reply:
Thanks for explaining, makes sense.